pygradle-plugin/src/test/groovy/com/linkedin/gradle/python/tasks/action/PipInstallActionTest.groovy

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2016 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.gradle.python.tasks.action
+
+import com.linkedin.gradle.python.exception.PipExecutionException
+import com.linkedin.gradle.python.extension.PythonDetailsTestDouble
+import com.linkedin.gradle.python.extension.VirtualEnvironment
+import com.linkedin.gradle.python.tasks.action.pip.PipActionHelpers
+import com.linkedin.gradle.python.tasks.action.pip.PipInstallAction
+import com.linkedin.gradle.python.tasks.exec.ExternalExecTestDouble
+import com.linkedin.gradle.python.util.DefaultEnvironmentMerger
+import com.linkedin.gradle.python.util.PackageSettings
+import com.linkedin.gradle.python.wheel.EmptyWheelCache
+import org.gradle.process.ExecSpec
+import org.gradle.testfixtures.ProjectBuilder
+import org.junit.Rule
+import org.junit.rules.TemporaryFolder
+import spock.lang.Specification
+
+import static com.linkedin.gradle.python.tasks.action.pip.PipActionHelpers.packageInGradleCache
+
+class PipInstallActionTest extends Specification {
```

Comment:
All of these tests just confirm that the arguments will be correctly passed if the action gets actually called.
They are however not a replacement for integration tests because they cannot actually prove that the action was ever called in real integration.